xbmc.py

```diff
@@ -350,23 +350,27 @@ def translatePath(path):  # NOSONAR
 
     """
 
+    def get_return_path(base_path, name, *segments):
```

Comment:
I usually don't like nested methods, but I understand the issue in `xbmc.py` as it is just a bunch of methods anyways. 

Could you include `.replace("/", os.sep)` in the `*[i for i in segments if i and i != '']`? That would make something like `*[i.replace("/", os.sep) for i in segments if i and i != '']`

And remove it from line 365, 369 and 373?

Reply:
Me neither but thought the nested methods are the pyhtonic way and thought that reusability is very low here justifying the nested method. 

Made the changes as requested so replacing all path seperators on the segments.

Reply:
Oh, the nested method here made sense. There was no need to move it. Is it too much to ask you to move it back? :)

Reply:
Sure moved back